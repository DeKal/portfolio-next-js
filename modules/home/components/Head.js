import React from 'react'
import NextHead from 'next/head'

const Head = () => (
  <NextHead>
    <title>PhatHo Portfolio and Website</title>
    <meta charSet="utf-8" />
  </NextHead>
)

export default Head
