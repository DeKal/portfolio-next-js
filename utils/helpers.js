import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
