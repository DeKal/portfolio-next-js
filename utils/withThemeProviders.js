import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
