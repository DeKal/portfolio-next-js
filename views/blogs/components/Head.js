import React from 'react'
import NextHead from 'next/head'

const Head = () => (
  <NextHead>
    <title>Coding with the Devil ;)</title>
    <meta charSet="utf-8" />
    <meta name="description" content="Coding with the Devil ;)" />
    <meta name="keywords" content="coding, blogs, diaries" />
  </NextHead>
)

export default Head
