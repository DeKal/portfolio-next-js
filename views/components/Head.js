import * as React from 'react'

const Head = () => (
  <head>
    <title>Portfolio | Phat Ho</title>
    <link rel="icon" href="/favicon.ico" />
    <link rel="canonical" href="https://phatho-folio.now.sh/" />
    <meta charSet="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <meta name="description" content="Dekal 's Portfolio" />
    <meta
      name="keywords"
      content="portfolio, dekal, software engineer, freelancer, cv, work"
    />
    <meta property="og:type" content="website" />
    <meta property="og:title" content="DeKal | Portfolio" />
    <meta property="og:description" content="Dekal's Portfolio" />
    <meta
      property="og:image"
      content="https://phatho-folio.now.sh/images/cover.jpg"
    />
    <meta
      property="og:image:secure_url"
      content="https://phatho-folio.now.sh/images/cover.jpg"
    />
    <meta property="og:url" content="https://phatho-folio.now.sh/" />
    <meta property="og:locale" content="en_VN" />
    <meta property="og:site_name" content="Phat Portfolio" />
    <meta
      name="google-site-verification"
      content="ofwBFRuFL3aycSJjDcrhc8hWEPKuJ7LkNCLUrsB0Sj4"
    />
    <meta name="robots" content="index,follow" />
    <meta name="googlebot" content="index,follow" />
    <script
      async=""
      src="https://www.googletagmanager.com/gtag/js?id=UA-102977389-1"
    ></script>
  </head>
)

export default Head
