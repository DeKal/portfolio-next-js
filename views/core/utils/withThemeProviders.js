import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
