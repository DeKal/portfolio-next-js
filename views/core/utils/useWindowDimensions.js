import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
