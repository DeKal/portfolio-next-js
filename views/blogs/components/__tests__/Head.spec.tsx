import * as React from 'react'
import { mountWithTheme } from '~/views/core/utils/withThemeProviders'
import Head from '~/views/blogs/components/Head'

describe('Head', () => {

  it('should render without throwing an error', function () {
    const wrap = mountWithTheme(<Head />)
    expect(wrap).toMatchSnapshot()
  })

})
