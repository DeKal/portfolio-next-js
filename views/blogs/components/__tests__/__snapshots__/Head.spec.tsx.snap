// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head should render without throwing an error 1`] = `
<Head>
  <Head>
    <Component
      handleStateChange={null}
      inAmpMode={false}
      reduceComponentsToState={[Function]}
    />
  </Head>
</Head>
`;
